### ISSUE TITLE
Potential application of this project

### ISSUE THREAD

**Issue by PikaMug** [Aetopia/NoSteamWebHelper#18]
Hello Aetopia,

Could I implore you to look into whether NoSteamWebHelper could include a workaround for this issue?
https://steamcommunity.com/discussions/forum/1/6725643788916599657/

Tl;dr: Recently, client version 1718305227 was pushed to the masses, and it appears to break Steam for anyone with multiple graphics options enabled. For example, if someone has both an AMD Radeon HD 7990 (primary) and Intel integrated graphics (secondary), they'd likely be affected. The exact symptoms are that Steam WebHelper crashes repeatedly, none of the client-suggested choices (like disabling GPU acceleration) work, and the problem cannot be remedied by any of the common solutions online.

The reason I ask is, setting `-no-browser` came up a lot and, after switching to integrated graphics, installing your project sped up my client considerably. Would it be out of scope for it to also undo whatever change was made to the web helper to bring this about?

This post from a separate thread is particularly relevant:
https://steamcommunity.com/discussions/forum/1/4333106230734794460/?ctp=2#c4343239957176490349

**Comment by Aetopia** [comment]
The reason why this issue gets fixed is due to the arguments NoSteamWebHelper launches Steam with:
- `-cef-single-process`
    Run CEF as a single process.

- `-cef-disable-breakpad`
       Disable crash reporting as well as the process responsible for it.

NoSteamWebHelper uses these arguments to restrict the WebHelper to a single process for easier process termination when enabling and disabling it.

- `-cef-disable-gpu-compositing`
  Same as disabling acclerated GPU WebViews.

This option seems to effectively solve this issue you mentioned.

For some or the other reason using GPU WebViews with NoSteamWebHelper would causes Steam crash on startup (You can't start Steam unless the WebHelper is running.).

**Comment by PikaMug** [comment]
Thanks for your reply. Unfortunately, `-cef-disable-gpu-compositing` does not solve it.

I should have better explained that NoSteamWebHelper does not currently fix the issue, but rather once the only known workaround is applied (disabling the primary graphics option) your project is helpful in speeding up WebHelper on the secondary (slow integrated graphics in my example).

Since it's known that the recent update is causing WebHelper to crash, I was wondering if you might try including fix for this in NoSteamWebHelper, if you feel it's adjacent.

**Comment by Aetopia** [comment]
I see.
What about trying to select what GPU, the WebHelper runs on?

If it doesn't run on the problematic GPU then there should be no problems.

**Comment by PikaMug** [comment]
> I see. What about trying to select what GPU, the WebHelper runs on?
> 
> If it doesn't run on the problematic GPU then there should be no problems.

Is that possible for an end user? Seems like a good solution, but I can't find anything about it online.

**Comment by Aetopia** [comment]
You can.
https://www.intel.com/content/www/us/en/support/articles/000090168/graphics.html

**Comment by PikaMug** [comment]
Ah, thought you meant via Steam. The summary in that link says "or integrated graphics" but the instructions are only for dedicated graphics cards. [Here](https://superuser.com/questions/1439471/how-can-i-force-my-dedicated-gpu-to-handle-all-applications-or-disable-my-integr) are steps I just tried to follow, but step 9 doesn't give me the integrated graphics as a choice. I'll keep trying and follow up.

**Comment by Aetopia** [comment]
Would you mind sending a screenshot of your GPU options?

**Comment by PikaMug** [comment]
Kind of you to assist (:

![gfx](https://github.com/Aetopia/NoSteamWebHelper/assets/2267126/116361a0-9fcc-4576-a46b-eed22bb36e42)

Apparently, the 'Power saving' choice is usually the iGPU, but not always.

**Comment by Aetopia** [comment]
Is this older hardware by any chance?
If it that is the case then reporting the issue to Valve is the only best choice or downgrading Steam.

**Comment by PikaMug** [comment]
It is older, but yes I agree that's the ideal solution. If you're not super interested in adding a workaround to NoSteamWebHelper that's understandable, but I appreciate you taking a look.

**Comment by Aetopia** [comment]
Closing this issue.
